src/fiatconnect/saga.ts

```diff
@@ -166,6 +169,64 @@ export function* handleSubmitFiatAccount({
         fiatType: quote.getFiatType(),
       })
     )
+
+    // If KYC is required, only proceed to Review screen if it's been approved.
+    // If any unexpected error is encountered, just show the pending screen,
+    // since the account has already been added. Otherwise, show the appropriate
+    // status screen.
+    const kycSchema = quote.getKycSchema()
+    if (kycSchema) {
+      try {
+        // If and only if KYC is required, we wait a bit to give KYC some time to process.
+        yield delay(KYC_WAIT_TIME_MILLIS)
+        const getKycStatusResponse = yield call(getKycStatus, {
+          providerInfo: quote.quote.provider,
+          kycSchemas: [kycSchema],
+        })
+        const fiatConnectKycStatus = getKycStatusResponse.kycStatus[kycSchema]
+        // const fiatConnectKycStatus = FiatConnectKycStatus.KycApproved as FiatConnectKycStatus
+        switch (fiatConnectKycStatus) {
+          case FiatConnectKycStatus.KycApproved:
+            yield put(submitFiatAccountKycApproved())
+            yield delay(500) // Allow user to admire green checkmark
+            break
+          // Denied, Expired, and Pending all fall through to the default case.
+          case FiatConnectKycStatus.KycDenied:
+            navigate(Screens.KycDenied, {
+              flow,
+              quote,
+              retryable: true, // TODO: Get this dynamically once IHL supports it
```

Comment:
what does IHL need to support for this to be dynamic? KYC deletion?

Reply:
[ACT-459](https://linear.app/valora/issue/ACT-459/walletihl-limit-number-of-kyc-retries-after-kyc-denied)